spring-cloud-gcp-core/src/main/java/org/springframework/cloud/gcp/core/Credentials.java

```diff
@@ -16,27 +16,26 @@
 
 package org.springframework.cloud.gcp.core;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 
 import org.springframework.core.io.Resource;
 
 /**
- * To be extended by property class subclasses and allow e.g., a "credentials.location" property
- * notation.
+ * Credentials configuration
  *
  * @author João Andre Martins
+ * @author Stephane Nicoll
  */
-public abstract class AbstractCredentialsProperty {
+public class Credentials {
 
-	private Resource location;
-	private List<String> scopes;
+	private List<String> scopes = new ArrayList<>();
 
-	public Resource getLocation() {
-		return this.location;
-	}
+	private Resource location;
 
-	public void setLocation(Resource location) {
-		this.location = location;
+	public Credentials(String... defaultScopes) {
+		this.scopes.addAll(Arrays.asList(defaultScopes));
```

Comment:
```
ArrayList(E[] array) {
 a = Objects.requireNonNull(array);
}
```
So, we need assertions do not allow end-users to specify `null` for the `defaultScopes`.

Or is it OK in the current Spring Boot to rely just on the `NPE`, @snicoll ?

Thanks

Reply:
We can add an assertion yes. It's not really a public API so I didn't bother